src/MolSim.cpp

```diff
@@ -28,77 +36,246 @@ void calculateV();
-  iterator = particles.begin();
+void print_help() {
+  std::cout << "Usage: MolSim [options]\n";
+  std::cout << "Options:\n";
+  std::cout << "  -h                 Show this help message\n";
+  std::cout << "  -o   <file_path>   Specify output file path\n";
+  std::cout << "  -i   <file_path>   Specify input file path\n";
+  std::cout
+      << "  -e   <end_time>    Specify how long the simulation should run\n";
+  std::cout << "  -d   <time_delta>  Specify time increments\n";
+  std::cout << "  -t                 Enable testing mode (Writes a file for "
+               "each iteration)\n";
+  std::cout << "  -x                 Output .xyz files instead of .vpu\n";
+}
 
+void calculateF() {
   for (auto &p1 : particles) {
+    std::array<double, 3> force_copy = p1.getF();
     for (auto &p2 : particles) {
-      // @TODO: insert calculation of forces here!
+      double f_x, f_y, f_z = 0;
+      double distance = std::sqrt(std::pow(p1.getX()[0] - p2.getX()[0], 2) +
+                                  std::pow(p1.getX()[1] - p2.getX()[1], 2) +
+                                  std::pow(p1.getX()[2] - p2.getX()[2], 2));
+      if (!(p1 == p2)) {
+        f_x = (p2.getX()[0] - p1.getX()[0]) * (p1.getM() * p2.getM()) /
+              pow(distance, 3);
+        f_y = (p2.getX()[1] - p1.getX()[1]) * (p1.getM() * p2.getM()) /
+              pow(distance, 3);
+        f_z = (p2.getX()[2] - p1.getX()[2]) * (p1.getM() * p2.getM()) /
+              pow(distance, 3);
+        p1.updateF(p1.getF()[0] + f_x, p1.getF()[1] + f_y, p1.getF()[2] + f_z);
+      }
+    }
+    p1.updateOldF(force_copy[0], force_copy[1], force_copy[2]);
+  }
+}
+
+void calculateF_new() {
+  // Store the current force as the old force and reset current to 0
+  for (auto &p1 : particles) {
+    p1.updateOldF(p1.getF()[0], p1.getF()[1], p1.getF()[2]);
+    p1.updateF(0, 0, 0);
+  }
+
+  // Simple force calculation formula (14)
+  for (auto it1 = particles.begin(); it1 != particles.end(); ++it1) {
+    for (auto it2 = ++it1; it2 != particles.end(); ++it2) {
+      Particle &p1 = *it1;
+      Particle &p2 = *it2;
+
+      double distance = std::sqrt(std::pow(p1.getX()[0] - p2.getX()[0], 2) +
+                                  std::pow(p1.getX()[1] - p2.getX()[1], 2) +
+                                  std::pow(p1.getX()[2] - p2.getX()[2], 2));
+
+      // Avoid division by zero
+      if (distance > 0) {
+        double f_x = (p2.getX()[0] - p1.getX()[0]) * (p1.getM() * p2.getM()) /
+                     pow(distance, 3);
+        double f_y = (p2.getX()[1] - p1.getX()[1]) * (p1.getM() * p2.getM()) /
+                     pow(distance, 3);
+        double f_z = (p2.getX()[2] - p1.getX()[2]) * (p1.getM() * p2.getM()) /
+                     pow(distance, 3);
+
+        p1.updateF(p1.getF()[0] + f_x, p1.getF()[1] + f_y, p1.getF()[2] + f_z);
+        // Newton's third law
+        p2.updateF(p2.getF()[0] - f_x, p2.getF()[1] - f_y, p2.getF()[2] - f_z);
+      }
```

Comment:
this is nice! but I guess it is not used right now in `MolSim.cpp`. But it seems correct to me. I would however use `distance > 1e-5` instead of just using 0.